### PR TITLE
Lvnod-69 create Dashboard component and Events List page

### PR DESCRIPTION
**Pull request by yuliannakorol**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-69)

## Code reviewers

- [ ] @AlbaMontis
- [ ] @shyyka
- [x] @avoytovich

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

## Summary of change

- added Dashboard component with tab sub-navigation
- added mock EventsList component that will be implemented in another task/pull request

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR

### CODE REVIEW

**Review by shyyka — DISMISSED** [review]

File: app/src/containers/Dashboard/Dashboard.js [hunk 1]
```diff
@@ -1,7 +1,52 @@
...
+                <Tab.Pane eventKey='third'>
+                    My guests will he here
+                </Tab.Pane>
+                <Tab.Pane eventKey='fourth'>
+                    Create new event coponent will be here
+                </Tab.Pane>
+            </Tab.Content>
+            </Col>
+        </Row>
+    </Tab.Container>
```
Comment (AlbaMontis) [inline app/src/containers/Dashboard/Dashboard.js]: Fix indentation.

File: app/src/containers/Events/EventsList.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { Link } from 'react-router';
+import { Table, Button } from 'react-bootstrap';
+
+export default class EventsList extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1>My Mock Events</h1>
+        <Table>
+            <thead>
```
Comment (AlbaMontis) [inline app/src/containers/Events/EventsList.js]: Use 2 spaces for indentation.

File: app/src/containers/Dashboard/Dashboard.js [hunk 3]
```diff
@@ -1,7 +1,52 @@
...
+
+  render() {
+    return (
+      <div>
+        <h1>Dashboard</h1>
+        <Tab.Container id='left-tabs-example' defaultActiveKey='first'>
+          <Row className='clearfix'>
+            <Col sm={4}>
+              <Nav bsStyle='pills' stacked>
+                <NavItem eventKey='first'>
```
Comment (AlbaMontis) [inline app/src/containers/Dashboard/Dashboard.js]: You can do something like:
 `['My Cabinet', 'My events'...].map((title, index) => 
  <NavItem eventKey={index}>
                   {title}
 </NavItem>
)`

File: app/src/containers/Dashboard/Dashboard.js [hunk 4]
```diff
@@ -1,7 +1,52 @@
...
+                  Invited
+                </NavItem>
+                <NavItem eventKey='fourth'>
+                  Create New Event
+                </NavItem>
+              </Nav>
+            </Col>
+            <Col sm={8}>
+              <Tab.Content animation>
+                <Tab.Pane eventKey='first'>
```
Comment (AlbaMontis) [inline app/src/containers/Dashboard/Dashboard.js]: The same as above.

File: app/src/containers/Events/EventsList.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { Link } from 'react-router';
```
Comment (avoytovich) [inline app/src/containers/Events/EventsList.js:2]: why did you import this module?

**Review by AlbaMontis — DISMISSED** [review]

**Review by avoytovich — DISMISSED** [review]

File: app/src/containers/Events/EventsList.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        <ListGroup>
+          <ListGroupItem> {eventNode}</ListGroupItem>
+        </ListGroup>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state, ownProps) => {
+  return {
```
Comment (AlbaMontis) [inline app/src/containers/Events/EventsList.js]: You can just '=>({prop: Val})' without return.

File: app/src/redux/EventsReducers.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
+const RETRIEVE_EVENTS = 'RETRIEVE_EVENTS';
+const RETRIEVE_EVENTS_SUCCESS = 'RETRIEVE_EVENTS_SUCCESS';
+const RETRIEVE_EVENTS_FAILURE = 'RETRIEVE_EVENTS_FAILURE';
+import  { API }  from './../helper/constants';
+import request from './../helper/request';
+
+export const retrieveEvents = () => {
+  let token = sessionStorage.getItem('token');
+
+  return (dispatch) => {
```
Comment (AlbaMontis) [inline app/src/redux/EventsReducers.js]: Useless ()

**Review by shyyka — DISMISSED** [review]

**Review by avoytovich — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by AlbaMontis — DISMISSED** [review]

File: app/src/containers/Dashboard/Dashboard.js [hunk 8]
```diff
@@ -1,7 +1,38 @@
...
+        <Tab.Container id='left-tabs-example' defaultActiveKey={0}>
+          <Row className='clearfix'>
+            <Col sm={4}>
+              <Nav bsStyle='pills' stacked>
+                {[
+                  'My Cabinet',
+                  'My Events',
+                  'Invitations',
+                  'Create New Event'
+                ].map((title, index) => <NavItem eventKey={index}> {title} </NavItem>)}
```
Comment (4asof4ik) [inline app/src/containers/Dashboard/Dashboard.js]: drop spaces {title}
check all PR

File: app/src/containers/Events/EventsList.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  componentWillMount() {
+    this.props.actions.retrieveEvents(this.state.events);
+    const data = this.props.events;
+    this.setState({ events: data.events });
+  }
+
+  render() {
+
```
Comment (4asof4ik) [inline app/src/containers/Events/EventsList.js]: drop

File: app/src/containers/Events/EventsList.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+    const id = this.props.id;
+    const eventNode = this.props.events.events.map((item) => {
```
Comment (4asof4ik) [inline app/src/containers/Events/EventsList.js]: events.events what is that?

File: app/src/containers/Events/EventsList.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+        <ListGroup>
+          <ListGroupItem> {eventNode}</ListGroupItem>
+        </ListGroup>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state) => ({
+
```
Comment (4asof4ik) [inline app/src/containers/Events/EventsList.js]: drop empty lines and 51,55

File: app/src/containers/Events/EventsList.js [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+          <ListGroupItem> {eventNode}</ListGroupItem>
+        </ListGroup>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state) => ({
+ 
+  events: state.events,
```
Comment (4asof4ik) [inline app/src/containers/Events/EventsList.js]: drop , check all PR

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by AlbaMontis — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by avoytovich — DISMISSED** [review]

File: app/src/containers/Events/EventsList.js [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+  constructor (props, context) {
+    super(props, context);
+    this.state = {
+      events: []
+    };
+  }
+
+  componentWillMount() {
+    this.props.actions.retrieveEvents(this.state.events);
+    const data = this.props.eventsList;
```
Comment (AlbaMontis) [inline app/src/containers/Events/EventsList.js]: Useless const.

File: app/src/containers/Events/EventsList.js [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+        <ListGroup>
+          <ListGroupItem>{eventNode}</ListGroupItem>
+        </ListGroup>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state) => ({
+  eventsList: state.eventsList
```
Comment (AlbaMontis) [inline app/src/containers/Events/EventsList.js]: state.eventsList.events

File: app/src/containers/Events/EventsList.js [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+        <h2>Events</h2>
+        <ListGroup>
+          <ListGroupItem>{eventNode}</ListGroupItem>
+        </ListGroup>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state) => ({
```
Comment (AlbaMontis) [inline app/src/containers/Events/EventsList.js]: Useless ();

File: app/src/containers/Events/EventsList.js [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+  eventsList: state.eventsList
+});
+
+const mapDispatchToProps = (dispatch) => ({
```
Comment (AlbaMontis) [inline app/src/containers/Events/EventsList.js]: Useless ();

**Review by avoytovich — DISMISSED** [review]

**Review by AlbaMontis — APPROVED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by AlbaMontis — DISMISSED** [review]

**Review by avoytovich — DISMISSED** [review]

File: app/src/redux/EventsReducers.js [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+export const retrieveEvents = () => {
+
+  return dispatch => {
+    return request()
+      .get(API.HOST + API.PORT + '/api/events')
+      .end((err, res) => {
+        if (err) {
+          dispatch({
+            type: RETRIEVE_EVENTS_FAILURE,
+            payload: err,});
```
Comment (4asof4ik) [inline app/src/redux/EventsReducers.js]: new line })

**Review by 4asof4ik — CHANGES_REQUESTED** [review]
Dashboard should be in components folder

**Pull request closed without merging**